### PR TITLE
Add Sagemaker examples

### PR DESCRIPTION
**Pull request by liveaverage**
- Example shim proxy layer to support running on Sagemaker
- Quick copy/paste cmds for AWS CLI
- Python helper for building, pushing, and deploying shimmed NIMs

### CODE REVIEW

**Comment by supertetelman** [comment]
Can we move the `sagemaker` directory into the `docs` directory?

File: sagemaker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM ${SRC_IMAGE}
+USER 0
+RUN apt-get update && apt-get install -y curl
+ENTRYPOINT ["sh", "-c", "curl -L https://bit.ly/nimshim-launch | bash -xe -s -- -c https://bit.ly/nimshim-caddy -e /opt/nim/start-server.sh"]
```
Comment (supertetelman) [inline sagemaker/Dockerfile:4]: This seems bad practice. Is there any way we can avoid curling a file at runtime and instead run this launcher locally?

Comment (liveaverage) [reply sagemaker/Dockerfile:4]: We could statically embed the caddy binary, launch.sh, etc. and allow specification of a custom config via env variable in the sagemaker model config, but the idea was to keep changes to the underlying NIM image as minimal as possible. If SM supported entrypoint overrides and running non-root I would've bypassed a shim image layer entirely (and exclusively provided a proxy layer via this method).

Comment (liveaverage) [reply sagemaker/Dockerfile]: Fixed with latest commit @supertetelman

**Comment by liveaverage** [comment]
Moved to dir `cloud-service-providers` per conversation

**Review by supertetelman — APPROVED** [review]
Thanks for updating the directory structure, looks good.

**Pull request merged**